viamonvif/service_test.go

```diff
@@ -213,6 +215,69 @@ func TestDoCommandPreview(t *testing.T) {
 		test.That(t, err.Error(), test.ShouldContainSubstring, "500: Internal Server Error")
 		test.That(t, result, test.ShouldBeNil)
 	})
+
+	t.Run("Test successful preview command with broken snapshot URI and valid streaming URI", func(t *testing.T) {
+		logger := logging.NewTestLogger(t)
+
+		bURL, err := base.ParseURL("rtsp://127.0.0.1:32512")
+		test.That(t, err, test.ShouldBeNil)
+		forma := &format.H264{
+			PayloadTyp:        96,
+			PacketizationMode: 1,
+			SPS: []uint8{
+				0x67, 0x64, 0x00, 0x15, 0xac, 0xb2, 0x03, 0xc1,
+				0x1f, 0xd6, 0x02, 0xdc, 0x08, 0x08, 0x16, 0x94,
+				0x00, 0x00, 0x03, 0x00, 0x04, 0x00, 0x00, 0x03,
+				0x00, 0xf0, 0x3c, 0x58, 0xb9, 0x20,
+			},
+			PPS: []uint8{0x68, 0xeb, 0xc3, 0xcb, 0x22, 0xc0},
+		}
+		h, closeFunc := viamrtsp.NewH264ServerHandler(t, forma, bURL, logger)
```

Comment:
if this test fails it won't close

```suggestion
		h, closeFunc := viamrtsp.NewH264ServerHandler(t, forma, bURL, logger)
		defer closeFunc()
```

Reply:
Yes thank you